ffi/diplomat/src/displaynames.rs

```diff
@@ -0,0 +1,137 @@
+    use crate::locale::ffi::ICU4XLocale;
+    use crate::provider::ffi::ICU4XDataProvider;
+    use alloc::boxed::Box;
+    use diplomat_runtime::DiplomatWriteable;
+    #[allow(unused_imports)] // feature-specific
+    use icu_displaynames::options::{DisplayNamesOptions, Fallback, LanguageDisplay};
+    use icu_displaynames::{LanguageDisplayNames, RegionDisplayNames};
+    use writeable::Writeable;
+
+    //  FFI version of `LanguageDisplayNames`.
+    #[diplomat::opaque]
+    #[diplomat::rust_link(icu::displaynames::LanguageDisplayNames, Struct)]
+    pub struct ICU4XLanguageDisplayNames(pub LanguageDisplayNames);
+
+    //  FFI version of `RegionDisplayNames`.
+    #[diplomat::opaque]
+    #[diplomat::rust_link(icu::displaynames::RegionDisplayNames, Struct)]
+    pub struct ICU4XRegionDisplayNames(pub RegionDisplayNames);
+
+    //  FFI version of `DisplayNamesOptions`.
+    #[diplomat::rust_link(icu::displaynames::options::DisplayNamesOptions, Struct)]
+    pub struct ICU4XDisplayNamesOptions {
+        /// The optional formatting style to use for display name.
+        pub style: ICU4XStyle,
+        /// The fallback return when the system does not have the
+        /// requested display name, defaults to "code".
+        pub fallback: ICU4XFallback,
+        /// The language display kind, defaults to "dialect".
+        pub language_display: ICU4XLanguageDisplay,
+    }
+
+    // FFI version of `Style`.
+    #[diplomat::rust_link(icu::displaynames::options::Style, Enum)]
+    pub enum ICU4XStyle {
+        Auto,
+        Narrow,
+        Short,
+        Long,
+        Menu,
+    }
+
+    // FFI version of `Fallback`.
+    #[diplomat::rust_link(icu::displaynames::options::Fallback, Enum)]
+    #[diplomat::enum_convert(Fallback, needs_wildcard)]
+    pub enum ICU4XFallback {
+        Code,
+        None,
+    }
+
+    // FFI version of `LanguageDisplay`.
+    #[diplomat::rust_link(icu::displaynames::options::LanguageDisplay, Enum)]
+    #[diplomat::enum_convert(LanguageDisplay, needs_wildcard)]
+    pub enum ICU4XLanguageDisplay {
+        Dialect,
+        Standard,
+    }
+
+    impl ICU4XLanguageDisplayNames {
+        /// Creates a new `LanguageDisplayNames` from locale data and an options bag.
+        #[diplomat::rust_link(
+            icu::displaynames::LanguageDisplayNames::try_new_unstable,
+            FnInStruct
+        )]
+        pub fn try_new_unstable(
+            provider: &ICU4XDataProvider,
+            locale: &ICU4XLocale,
+            options: ICU4XDisplayNamesOptions,
+        ) -> Result<Box<ICU4XLanguageDisplayNames>, ICU4XError> {
+            let locale = locale.to_datalocale();
+            let options = DisplayNamesOptions::from(options);
+
+            Ok(Box::new(ICU4XLanguageDisplayNames(
+                LanguageDisplayNames::try_new_unstable(&provider.0, &locale, options)?,
+            )))
+        }
+
+        // Returns the locale specific display name of a language for a given string.
```

Comment:
Here and below: Please document that the function returns an empty string in place of the None value.

Reply:
Yep done.

Reply:
Please use doc comments (three slashes)

Reply:
Done and regenerated ffi bindings to make sure that the comment is visible.